app/views/catalog/index.html.erb

```diff
@@ -0,0 +1,26 @@
+<% # TRLN override of BL8 catalog/index view %>
```

Comment:
This does indeed work, but overriding the entire BL core index template comes at a cost. It's probably better in this case to just override/extend BL's view for the sidebar component specifically (https://github.com/projectblacklight/blacklight/blob/release-8.x/app/components/blacklight/search/sidebar_component.html.erb). E.g., if you make a local copy of only that file, then put this at the top, it'd work:
```
<%= render 'catalog/local_filter',
           local_button_id: 'toggle-local-btn-top',
           trln_button_id: 'toggle-trln-btn-top' %>
```
It's a bit subjective, so I think worth team weigh-in on how we want to approach this kind of refactor for BL8.

Reply:
Okay, sounds good. I tried to create a new "local_filter" component to replace that partial, but found that the methods and variables that were available to the view partial were undefined when used in the component. You're saying to remove the "local_filter" partial and move that markup/logic into an override of the new sidebar component? I think it might be a little more complicated than that as the new sidebar component is not a 1-to-1 replacement for the old sidebar partial.

Reply:
I would keep the `_local_filter.html.erb` partial how you have it. Copy the https://github.com/projectblacklight/blacklight/blob/release-8.x/app/components/blacklight/search/sidebar_component.html.erb file into this engine at the same path. Prepend that snippet to the top, annotated with comments like you've done elsewhere.

You probably don't need to pass `search_state` to the partial. If you do, the component can still call that method but with `helpers.search_state`, i.e.:
```
<%= render 'catalog/local_filter',
           search_state: helpers.search_state,
           local_button_id: 'toggle-local-btn-top',
           trln_button_id: 'toggle-trln-btn-top' %>
```

Reply:
I found that i had to also include the 'sidebar_component.rb' to get the override to work, as well as updating the configuration to use the override.